src/crypto/tests/TestChipCryptoPAL.cpp

```diff
@@ -1226,66 +1238,65 @@ static void TestECDH_EstablishSecret(nlTestSuite * inSuite, void * inContext)
     out_secret2.Bytes()[0] = 1;
 
     CHIP_ERROR error = CHIP_NO_ERROR;
-    NL_TEST_ASSERT(inSuite,
-                   memcmp(out_secret1.ConstBytes(), out_secret2.ConstBytes(), out_secret1.Capacity()) !=
-                       0); // Validate that buffers are indeed different.
+    EXPECT_NE(memcmp(out_secret1.ConstBytes(), out_secret2.ConstBytes(), out_secret1.Capacity()),
+              0); // Validate that buffers are indeed different.
 
     error = keypair2.ECDH_derive_secret(keypair1.Pubkey(), out_secret1);
-    NL_TEST_ASSERT(inSuite, error == CHIP_NO_ERROR);
+    EXPECT_EQ(error, CHIP_NO_ERROR);
 
     error = keypair1.ECDH_derive_secret(keypair2.Pubkey(), out_secret2);
-    NL_TEST_ASSERT(inSuite, error == CHIP_NO_ERROR);
+    EXPECT_EQ(error, CHIP_NO_ERROR);
 
     bool signature_lengths_match = out_secret1.Length() == out_secret2.Length();
-    NL_TEST_ASSERT(inSuite, signature_lengths_match);
+    EXPECT_TRUE(signature_lengths_match);
 
     bool signatures_match = (memcmp(out_secret1.ConstBytes(), out_secret2.ConstBytes(), out_secret1.Length()) == 0);
-    NL_TEST_ASSERT(inSuite, signatures_match);
+    EXPECT_TRUE(signatures_match);
 }
 
 #if CHIP_CRYPTO_OPENSSL
-static void TestAddEntropySources(nlTestSuite * inSuite, void * inContext)
+TEST_F(TestChipCryptoPAL, TestAddEntropySources)
 {
-    HeapChecker heapChecker(inSuite);
+    HeapChecker heapChecker;
     CHIP_ERROR error = add_entropy_source(test_entropy_source, nullptr, 10);
-    NL_TEST_ASSERT(inSuite, error == CHIP_NO_ERROR);
+    EXPECT_EQ(error, CHIP_NO_ERROR);
     uint8_t buffer[5];
-    NL_TEST_ASSERT(inSuite, DRBG_get_bytes(buffer, sizeof(buffer)) == CHIP_NO_ERROR);
+    EXPECT_EQ(DRBG_get_bytes(buffer, sizeof(buffer)), CHIP_NO_ERROR);
 }
 #endif
 
 #if CHIP_CRYPTO_MBEDTLS
-static void TestAddEntropySources(nlTestSuite * inSuite, void * inContext)
+TEST_F(TestChipCryptoPAL, TestAddEntropySources)
 {
-    HeapChecker heapChecker(inSuite);
+    HeapChecker heapChecker;
     CHIP_ERROR error = add_entropy_source(test_entropy_source, nullptr, 10);
-    NL_TEST_ASSERT(inSuite, error == CHIP_NO_ERROR);
+    EXPECT_EQ(error, CHIP_NO_ERROR);
     uint8_t buffer[5];
     uint32_t test_entropy_source_call_count = gs_test_entropy_source_called;
-    NL_TEST_ASSERT(inSuite, DRBG_get_bytes(buffer, sizeof(buffer)) == CHIP_NO_ERROR);
+    EXPECT_EQ(DRBG_get_bytes(buffer, sizeof(buffer)), CHIP_NO_ERROR);
     for (int i = 0; i < 5000 * 2; i++)
     {
         (void) DRBG_get_bytes(buffer, sizeof(buffer));
     }
-    NL_TEST_ASSERT(inSuite, gs_test_entropy_source_called > test_entropy_source_call_count);
+    EXPECT_GT(gs_test_entropy_source_called, test_entropy_source_call_count);
 }
 #endif
 
 #if CHIP_CRYPTO_PSA
-static void TestAddEntropySources(nlTestSuite * inSuite, void * inContext) {}
+TEST_F(TestChipCryptoPAL, TestAddEntropySources)
```

Comment:
can we just delete these? I think they may have existed to make sure that the NL-unittest list contained some values that are valid always, however for gtest it looks like declaration via macro is marking the test as existing, so we do not need empty/noop entries.

Reply:
You are right; I've removed empty tests